src/Http/Controllers/QuestionAdminApiController.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace EscolaLms\Questionnaire\Http\Controllers;
+
+use EscolaLms\Core\Http\Controllers\EscolaLmsBaseController;
+use EscolaLms\Questionnaire\Http\Controllers\Contracts\QuestionAdminApiContract;
+use EscolaLms\Questionnaire\Http\Requests\QuestionCreateRequest;
+use EscolaLms\Questionnaire\Http\Requests\QuestionDeleteRequest;
+use EscolaLms\Questionnaire\Http\Requests\QuestionListingRequest;
+use EscolaLms\Questionnaire\Http\Requests\QuestionReadRequest;
+use EscolaLms\Questionnaire\Http\Requests\QuestionUpdateRequest;
+use EscolaLms\Questionnaire\Http\Resources\QuestionResource;
+use EscolaLms\Questionnaire\Models\Question;
+use EscolaLms\Questionnaire\Repository\Contracts\QuestionRepositoryContract;
+use Illuminate\Contracts\Support\Renderable;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Auth;
+
+class QuestionAdminApiController extends EscolaLmsBaseController implements QuestionAdminApiContract
+{
+    private QuestionRepositoryContract $questionRepository;
+
+    public function __construct(QuestionRepositoryContract $questionRepository)
+    {
+        $this->questionRepository = $questionRepository;
+    }
+
+    public function list(QuestionListingRequest $request): JsonResponse
+    {
+        try {
+            $questions = $this->questionRepository->searchAndPaginate();
+            return $this->sendResponseForResource(
+                QuestionResource::collection($questions),
+                "Question list retrieved successfully"
+            );
+        } catch (Renderable $e) {
+            return $this->sendError($e->getMessage());
+        }
+    }
```

Comment:
![image](https://user-images.githubusercontent.com/214608/151951254-a361c389-7ead-4de2-8916-515e32332546.png)

what kind of error can appear here 

if error appears then laravel event handler will shod proper error 

so removing all ` } catch (Renderable $e) {` will increase just code coverage and rest errors will be caught by parent exception handler 

so, remove all `try {} catch {}` blocks from controllers

Reply:
done